application/Validation.go

```diff
@@ -0,0 +1,46 @@
+package application
+
+import (
+	"authz/domain"
+	"regexp"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/golang/glog"
+)
+
+var validatorInstance *validator.Validate
+
+// ValidateEvent performs validation on the provided event struct and returns true if the struct is valid, else it returns false and an error message object
+func ValidateEvent(evt interface{}) (bool, error) {
+	if validatorInstance == nil {
+		initializeValidator() //TODO: move to startup?
+	}
+
+	err := validatorInstance.Struct(evt)
+
+	if err != nil {
+		errors, ok := err.(validator.ValidationErrors)
+		if !ok {
+			glog.Errorf("Failed to validate message %+v. Error: %+v", evt, err)
+			return false, nil
```

Comment:
I was thinking it would be good to have `ValidateEvent` return just `err` like `validatorInstance`, and test on `err` rather than `ok`. Not sure of the branch returning `(ok = false, err = nil)` makes sense to me. Feels like it must be some kind of error.